tools/tests/switch/test.js

```diff
@@ -0,0 +1,110 @@
+
+    return new Promise((resolve) => {
+        waitForStyles(resolve, 4);
+    });
+};
+
+describe('Switch Component', () => {
+    afterEach(() => {
+        // Since we don't want to replace the whole content of the body using
+        // innerHtml setter, we query only the current custom element and we replace
+        // it with a new one; this is needed because the specs are executed in a random
+        // order and sometimes the component might be left in a state that is not
+        // ready for testing
+        let switchToggle = document.querySelector('gameface-switch');
+
+        if (switchToggle) {
+            switchToggle.parentElement.removeChild(switchToggle);
+        }
+    });
+
+    it('Should be rendered', async () => {
+        await loadSwitch({});
+        const switchToggle = document.querySelector('gameface-switch');
+        assert.exists(switchToggle, 'Switch component is added to DOM');
+    });
+
+    it('Should be checked', async () => {
+        await loadSwitch({ checked: true });
+        const switchToggle = document.querySelector('gameface-switch');
+        const checkedSwitch = switchToggle.querySelector('.switch-toggle-checked');
+        const checkedHandle = switchToggle.querySelector('.switch-toggle-handle-checked');
+        assert.exists(checkedSwitch, 'Switch is checked');
+        assert.exists(checkedHandle, 'Switch handle is checked');
+    });
+
+    it('Should be disabled', async () => {
+        await loadSwitch({ disabled: true });
+        const switchToggle = document.querySelector('gameface-switch');
+        const disabledSwitch = switchToggle.querySelector('.switch-toggle-disabled');
+        assert.exists(disabledSwitch, 'Switch is disabled');
+    });
+
+    it('Should be inset', async () => {
+        await loadSwitch({ type: 'inset' });
+        const switchToggle = document.querySelector('gameface-switch');
+        const insetSwitch = switchToggle.querySelector('.switch-toggle-inset');
+        assert.exists(insetSwitch, 'Switch is inset');
+    });
+
+    it('Should have text inside', async () => {
+        await loadSwitch({ type: 'text-inside' });
+        const switchToggle = document.querySelector('gameface-switch');
+        assert.equal(switchToggle.firstChild.childElementCount, 1);
+    });
+
+    it('Should have correct labels', async () => {
+        const checkText = 'On';
+        const uncheckText = 'Off';
+        await loadSwitch({ checkText, uncheckText });
+        const switchToggle = document.querySelector('gameface-switch');
+        const checkedTextElement = switchToggle.querySelector('.switch-toggle-true');
+        const uncheckedTextElement = switchToggle.querySelector('.switch-toggle-false');
+        assert.equal(checkedTextElement.textContent.replace(/[\n\r]+|[\s]{2,}/g, ' ').trim(), checkText);
+        assert.equal(uncheckedTextElement.textContent.replace(/[\n\r]+|[\s]{2,}/g, ' ').trim(), uncheckText);

```

Comment:
Can we explain the regular expression in a comment, please?

Reply:
And for the white space, it runs 2 times in total.